Source/API/Extras/CBLIncrementalStore.m

```diff
@@ -346,255 +348,395 @@ -(BOOL)loadMetadata:(NSError **)outError
     return success;
 }
 
-- (id)executeRequest:(NSPersistentStoreRequest *)request withContext:(NSManagedObjectContext*)context error:(NSError **)outError
+- (BOOL)updateInverseParentKeyForRelationship:(NSRelationshipDescription *)relationship
+                                 withParentId:(NSString *)parentId
+                             forManagedObject:(NSManagedObject *)object
+                                     outRevId:(NSString **)outRevId
+                                     outError:(NSError **)outError
 {
-    if (request.requestType == NSSaveRequestType) {
-        
-        NSSaveChangesRequest *save = (NSSaveChangesRequest*)request;
-        
-#ifdef DEBUG_DETAILS
-        NSLog(@"[tdis] save request: ---------------- \n"
-              "[tdis]   inserted:%@\n"
-              "[tdis]   updated:%@\n"
-              "[tdis]   deleted:%@\n"
-              "[tdis]   locked:%@\n"
-              "[tids]---------------- ", [save insertedObjects], [save updatedObjects], [save deletedObjects], [save lockedObjects]);
-#endif
-        
-        // TODO: Check if using the CouchbaseLite transaction mechanism makes sense here.
-        
-        NSError *error;
-        
-        NSMutableSet *changedEntities = [NSMutableSet setWithCapacity:[save insertedObjects].count];
-        
-        // Objects that were inserted...
-        for (NSManagedObject *object in [save insertedObjects]) {
-            NSDictionary *contents = [self _couchbaseLiteRepresentationOfManagedObject:object withCouchbaseLiteID:YES];
-            
-            CBLDocument *doc = [self.database documentWithID:[object.objectID couchbaseLiteIDRepresentation]];
-            CBLUnsavedRevision *revision = [doc newRevision];
-            [revision.properties setValuesForKeysWithDictionary:contents];
-            
-            
-            // add attachments
-            NSDictionary *propertyDesc = [object.entity propertiesByName];
+    CBLDocument *doc = [self.database documentWithID:[object.objectID couchbaseLiteIDRepresentation]];
+    if (!doc) return YES;
```

Comment:
This will never happen — documentWithID: always returns a document even if it doesn't exist yet. To check for a nonexistent document, test for `doc.currentRevision == nil`.